src/librbd/cache/ImageCache.h

```diff
@@ -53,6 +53,8 @@ struct ImageCache {
 
   virtual void invalidate(Context *on_finish) = 0;
   virtual void flush(Context *on_finish) = 0;
+
+  static void discard_cache(ImageCtxT &image_ctx, Context *ctx);
```

Comment:
Nit: `ImageCache` is effectively dead and will be removed from the source.  Just add this helper to `WriteLogImageDispatch`

Reply:
@dillaman How about that we put this function in librbd/cache/Util.*? Currently, WriteLogImageDispatch is only built when WITH_RWL is ON.   
Or I have to add #ifdef(WITH_RWL) in librbd/internal.cc.